src/Compilers/CSharp/Portable/Symbols/Source/SourceMethodSymbol.cs

```diff
@@ -255,6 +256,11 @@ protected void SetReturnsVoid(bool returnsVoid)
             this.flags.ReturnsVoid = returnsVoid;
         }
 
+        protected void SetIsExtensionMethod(bool isExtensionMethod)
```

Comment:
From what I can see. This is just storing information about `SourceMethodSymbol` in a compact way. Things that can be calculated at once during the symbol constructor like `MethodKind` don't have setters. Things that require binding for its value like `ReturnsVoid` have both setters and getters. They are initialized with dummy values then set later during `MethodChecks` to its correct value.
Please see `Flags.ReturnsVoid` and `SetReturnsVoid()`.

---
In reply to: [122884712](https://github.com/dotnet/roslyn/pull/19703#discussion_r122884712) [](ancestors = 122884712)